src/pytorch_lightning/_graveyard/cloud_io.py

```diff
@@ -0,0 +1,23 @@
+import pickle
+import warnings
+from copy import deepcopy
+from typing import Any
+
+
+class RedirectingUnpickler(pickle.Unpickler):
+    def find_class(self, module: str, name: str) -> Any:
+        if module.startswith("pytorch_" + "lightning"):
+            # for the standalone package this won't do anything,
+            # for the unified mirror package it will redirect the imports
+
+            old_module = deepcopy(module)
+            module = "pytorch_lightning" + module[len("pytorch_" + "lightning") :]
+
+            # this warning won't trigger for standalone as these imports are identical
+            if module != old_module:
+                warnings.warn(f"Redirecting import of {module}.{name} to pytorch_lightning.{name}")
+
+        return super().find_class(module, name)
+
+
+pickle.Unpickler = RedirectingUnpickler  # type: ignore
```

Comment:
(not for this PR) I'm not sure the graveyard is the right place for this code.

Could it be added as a migration piece in the future? ref: #15237 

If we did, then maybe we can restore this `pickle.Unpickler` override back to the original

Reply:
Exactly my thoughts. I didn't want to interfer with existing checkpoint PRs however. I chose graveyard because it is for legacy checkpoints (we don't support pickling anymore) and graveyard was where we had patching of modules anyways.

I'd prefer to move it to checkpointing after #15237 .